data/__init__.py

```diff
@@ -78,11 +78,13 @@ def __init__(self, opt):
         self.opt = opt
         dataset_class = find_dataset_using_name(opt.dataset_mode)
         self.dataset = dataset_class(opt)
-        print("dataset [%s] was created" % type(self.dataset).__name__)
+        if rank==0:
+            print("dataset [%s] was created" % type(self.dataset).__name__)
+        sampler=data.distributed.DistributedSampler(self.dataset, shuffle=not opt.serial_batches)
```

Comment:
I think we should only use DistributedSampler if we are in a distributed context